### PR TITLE
mariadb: add missing deps and depexts

### PR DESCRIPTION
**Pull request by damiendoligez**

### CODE REVIEW

**Comment by yallop** [comment]
There's [still](https://travis-ci.org/ocaml/opam-repository/jobs/157655828) some kind of `depext`/linking problem:

```
# /usr/bin/ld: cannot find -lmysqlclient
```

**Comment by damiendoligez** [comment]
I found another one, but I still seem to be missing some, and I don't have the tools to debug this completely.

The OSX one is correct, AFAICT.

**Comment by yallop** [comment]
@andrenth, do you know which system packages are needed?

File: packages/mariadb/mariadb.0.5.0/opam [hunk 1]
```diff
@@ -15,5 +15,12 @@ depends: [
   "ocamlfind" {build}
   "ctypes" {>= "0.7.0"}
   "ctypes-foreign" {>= "0.4.0"}
+  "async"
+  "lwt"
```
Comment (dsheets) [inline packages/mariadb/mariadb.0.5.0/opam]: It *requires* both??

Comment (damiendoligez) [inline packages/mariadb/mariadb.0.5.0/opam]: Yes. It builds its example files, which include both Async examples and Lwt examples.

**Comment by dsheets** [comment]
Some of these depext questions were answered in the original #7222.

**Comment by andrenth** [comment]
I don't think the library should require either lwt or async, since they're not really dependencies. People interested in building The examples can clone the fit repository.

The libmariadbclient-dev packages in Debian and Ubuntu don't support the nonblocking calls, which is why I didn't add the depexts. You need to install the packages from mariadb.org.

**Comment by damiendoligez** [comment]
>I don't think the library should require either lwt or async, since they're not really dependencies. People interested in building The examples can clone the fit repository.

Then we should change the build instructions because calling just `make` will attempt to build the examples.

**Comment by andrenth** [comment]
Hmm weird, both flags in the _oasis file default to false...

**Comment by andrenth** [comment]
I've reorganized the examples structure to avoid this problem and released version 0.5.1 (#7408).

**Comment by damiendoligez** [comment]
Note: the travis failures are due to #7391. I've restarted the builds.

**Comment by yallop** [comment]
I think the remaining problem is a depext version problem, and the example/dependency issue is resolved in 0.5.1, so I'm merging this.

**Pull request merged**